pcdsdevices/sqr1.py

```diff
@@ -0,0 +1,280 @@
+        **kwargs,
+    ):
+        self._axis = axis.value
+        self._sync_setpoints = sync_setpoints
+        super().__init__(prefix, **kwargs)
+
+    def move(
+        self,
+        position: float,
+        wait: bool = True,
+        timeout: float = None,
+        moved_cb: typing.Callable | None = None,
+        sync_enable: bool = True,
+    ):
+        if sync_enable and self._sync_setpoints:
+            self._sync_setpoints()
+        return super().move(position, wait, timeout, moved_cb)
+
+
+class SQR1(Device):
+    """
+    This class represents the entire square-one tri-sphere motion system,
+    which consists of multiple axes (X, Y, Z, rX, rY, rZ) that can be
+    controlled individually. It aggregates instances of the `SQR1Axis` class
+    for each axis and provides methods for multi-axis movement and
+    stopping the motion.
+
+    Attributes:
+        x (SQR1Axis): An instance of the `SQR1Axis` class representing
+        the X-axis.
+        y (SQR1Axis): An instance of the `SQR1Axis` class representing
+        the Y-axis.
+        z (SQR1Axis): An instance of the `SQR1Axis` class representing
+        the Z-axis.
+        rx (SQR1Axis): An instance of the `SQR1Axis` class representing
+        the rX-axis.
+        ry (SQR1Axis): An instance of the `SQR1Axis` class representing
+        the rY-axis.
+        rz (SQR1Axis): An instance of the `SQR1Axis` class representing
+        the rZ-axis.
+        stop_signal (EpicsSignal): An EPICS signal for stopping the motion
+        of the entire tri-sphere motion system.
+
+    Args:
+        prefix (str): The EPICS PV prefix for the motion system.
+        name (str): The name for this device.
+        kind (str or None, optional): The kind of device
+        (e.g., 'detector', 'motor').
+        read_attrs (list or None, optional): List of attribute names to be
+        read when reading this device.
+        configuration_attrs (list or None, optional): List of attribute names
+        to be configured when configuring this device.
+        parent (Device or None, optional): The parent device, if applicable.
+        **kwargs: Additional keyword arguments to pass to the base
+        class constructor.
+
+    Methods:
+        sync_setpoints(): Synchronize the setpoints of all axes to their
+        respective readback values.
+        multi_axis_move(x_sp=None,
+                        y_sp=None,
+                        z_sp=None,
+                        rx_sp=None,
+                        ry_sp=None,
+                        rz_sp=None,
+                        wait=True,
+                        timeout=10.0):Move multiple axes simultaneously
+        to specified setpoints.
+        stop(): Stop the motion of the entire tri-sphere motion system.
+
+    Example:
+        >>> tri_sphere = SQR1(prefix="SQR1:", name="tri_sphere")
+        >>> tri_sphere.multi_axis_move(x_sp=1.0, y_sp=2.0, z_sp=3.0)
+        >>> tri_sphere.stop()
```

Comment:
These examples render really well in the html docs, we should do this more often :+1:
![image](https://github.com/pcdshub/pcdsdevices/assets/10647860/06a7645f-1f81-4d8e-a51e-0371b2eb2c5f)